root/usr/share/firewall4/templates/ruleset.uc

```diff
@@ -216,7 +222,9 @@ table inet fw4 {
 	chain input_{{ zone.name }} {
 {%  fw4.includes('chain-prepend', `input_${zone.name}`) %}
 {%  for (let rule in fw4.rules(`input_${zone.name}`)): %}
-		{%+ include("rule.uc", { fw4, rule }) %}
+	{%- if (rule.log): %}
+		{%+ include("rule.uc", { fw4, rule, log: true }) %} {%- endif %}
```

Comment:
Why not `log: rule.log` in one line?

Reply:
"log" does not mean "do the action and log" but "do the log action". That's why "log: rule.log" is not enough.

Reply:
I changed this logic quite a bit. I'll set this as resolved. Now the rule duplication happens inside each template.